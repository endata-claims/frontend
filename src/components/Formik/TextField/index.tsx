import React from 'react'

import { Field } from 'formik'
import { TextField } from 'formik-material-ui'
import { TextFieldProps } from '@material-ui/core'
import { InputAdornment } from '@material-ui/core'

export type FormikTextFieldProps = TextFieldProps & {
  name: string

  startAdornment?: React.ReactElement
  endAdornment?: React.ReactElement
  readOnly?: boolean
}

const FormikTextField: React.FC<FormikTextFieldProps> = ({ startAdornment, endAdornment, InputProps, ...props }) => {
  return (
    <Field component={TextField}
      {...props}
      InputProps={{
        ...InputProps,
        startAdornment: !startAdornment ? null : (
          <InputAdornment position='start'>
            {startAdornment}
          </InputAdornment>
        ),
        endAdornment: !endAdornment ? null : (
          <InputAdornment position='end'>
            {endAdornment}
          </InputAdornment>
        ),
      }}
    />
  )
}
export default FormikTextField