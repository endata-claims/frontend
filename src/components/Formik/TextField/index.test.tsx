import React from 'react'
import { shallow } from 'enzyme'
import { Formik, Form } from 'formik'
import YouTubeIcon from '@material-ui/icons/YouTube'

import TextField from '.'

describe('TextField tests', () => {
  const renderWrapper = (props = {}) => shallow(
    <Formik initialValues={{ text: 'some text' }}
      onSubmit={values => console.log(values)}
    >
      <Form>
        <TextField label='Text field' name='text' {...props} />
      </Form>
    </Formik>
  )

  describe('snapshots', () => {
    it('should render with label', () => {
      expect(renderWrapper()).toMatchSnapshot()
    })
    it('should render with icon', () => {
      expect(renderWrapper({
        startAdornment: <YouTubeIcon />
      })).toMatchSnapshot()
    })
    it('should render with right icon', () => {
      expect(renderWrapper({
        endAdornment: <YouTubeIcon />
      })).toMatchSnapshot()
    })
  })
})