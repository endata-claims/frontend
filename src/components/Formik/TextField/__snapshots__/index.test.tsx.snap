// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField tests snapshots should render with icon 1`] = `ShallowWrapper {}`;

exports[`TextField tests snapshots should render with label 1`] = `ShallowWrapper {}`;

exports[`TextField tests snapshots should render with right icon 1`] = `ShallowWrapper {}`;
